src/rest/RequestHandler.js

```diff
@@ -223,8 +254,22 @@ class RequestHandler {
     if (res.status >= 400 && res.status < 500) {
       // Handle ratelimited requests
       if (res.status === 429) {
-        // A ratelimit was hit - this should never happen
+        // A ratelimit was hit - this should only happen if client is already rate limited when connecting
```

Comment:
Changing this comment is not accurate, 429s should never happen. At MOST I'd say changing it to `this could happen`...

Reply:
really? I reached this very point when the client was already rate limited when connecting so I figured this was the reason one might get there

Reply:
The only ratelimit you *could* get is on `/gateway/bot` when identifying too fast

Reply:
Well I do hit 429s. This is with the current master branch

![image](https://user-images.githubusercontent.com/45497981/119261340-4088c500-bbd7-11eb-803b-42346a3f0575.png)

code:
```js
'use strict';

const Discord = require('..');

const client = new Discord.Client({
  intents: Discord.Intents.NON_PRIVILEGED,
});

client.once('ready', async () => {
  const chan = client.channels.cache.get('800884361733865522');
  while (true) {
    console.log('Editing channel...');
    await chan.edit({ name: Math.random().toString(36) });
  }
});

client.login('TOKEN'');
```

Reply:
Wait, hold on, you're telling me that on master, this code causes it to reply with a node-fetch Response? Are you sure?

Reply:
no, I added `console.log(res)` right after the `if (res.status === 429) {` (the if statement this thread is under), that is why it's logging the node fetch response. The rest of the djs code is the current master branch, the console.log is the only addition I made

Reply:
Oh, that error is not unexpected sadly. Discord has a very strict sub-limit for channel name and topic changes, which can explain why you hit a 429 so fast. :(

Reply:
so 429s should generally not happen, just on some specific routes that have these strict date limits? Is there not a way to catch these too and avoid rats limits or are the headers inaccurate here?

I will revert the comment, but it should be changed to describe these edge cases then

Reply:
These sub limits are:
- Not listed in the headers
- Separate from the headers
- Requests that don't trigger these sub limits can still be sent

Avoiding these is currently something we don't have, as we'd have to implement an edge case for each sub limit there is/will be. Implementation, if any, is to be done :D
Personally, I don't think we should change the comment to mention that, as it's invisible for most users anyways